addons/udes_stock/models/stock_location.py

```diff
@@ -134,6 +134,48 @@ def _domain_height_category(self):
         see u_countable field.""",
     )
 
+    u_heatmap_data_updated = fields.Datetime(
+        string="Heatmap data updated",
+        help="""
+            The date when the location name, co-ordinates or size was last changed. 
+        """,
+        default=fields.Datetime.now,
+    )
+
+    def set_u_heatmap_data_updated(self, vals):
+        """
+        Iterate over locations in `self`, checking if any heatmap fields
+        have been written to with updated values, and if so - dispatch to write()
+        with a bypass in context (to avoid recursion) to update `u_heatmap_data_updated`
+        with the current date & time.
+
+        :param: vals: dict() of vals from `write()`
+        """
+        StockLocation = self.env["stock.location"]
+        locations_to_update = StockLocation.browse()
+        for location in self:
```

Comment:
Might be worth checking if any of the heatmap fields are in `vals.keys()` before looping through.

Reply:
good shout